addons/stock/tests/test_move.py

```diff
@@ -4602,6 +4602,36 @@ def test_immediate_validate_9_tracked_move_with_0_qty_done(self):
         immediate_wizard_form.process()
         self.assertEqual(receipt.state, 'done')
 
+    def test_immediate_validate_10_tracked_move_without_backorder(self):
+        """
+            Create a picking for a tracked product, validate it as an
+            immediate transfer, and ensure that the backorder wizard is
+            not triggered when the qty is reserved.
+        """
+        picking_type_receipt = self.env.ref('stock.picking_type_internal')
```

Comment:
```suggestion
        picking_type_internal = self.env.ref('stock.picking_type_internal')
```
Same for the picking's variable=
```python
        internal_transfer = picking_form.save()
```